core/trino-main/src/main/java/io/trino/sql/planner/planprinter/JsonRenderer.java

```diff
@@ -54,6 +83,88 @@ private JsonRenderedNode renderJson(PlanRepresentation plan, NodeRepresentation
                         .collect(toImmutableList()));
     }
 
+    public Map<Integer, List<DistributedPlanRepresentation>> render(PlanPrinter planPrinter, boolean verbose, Integer level)
+    {
+        PlanRepresentation plan = planPrinter.getPlanRepresentation();
+        NodeRepresentation node = plan.getRoot();
+        Map<Integer, List<DistributedPlanRepresentation>> distributedPlanRepresentationMap = new HashMap<>();
+        distributedPlanRepresentationMap.computeIfAbsent(level, i -> new ArrayList<>()).add(getPlanRepresentation(plan, node, verbose, level));
+        return distributedPlanRepresentationMap;
+    }
+
+    public DistributedFragmentRepresentation formatFragment(
+            Function<TableScanNode, TableInfo> tableInfoSupplier,
+            Map<DynamicFilterId, DynamicFilterDomainStats> dynamicFilterDomainStats,
+            ValuePrinter valuePrinter,
+            PlanFragment planFragment,
+            Optional<StageInfo> stageInfo,
+            Optional<Map<PlanNodeId, PlanNodeStats>> planNodeStats,
+            TypeProvider typeProvider,
+            boolean verbose)
+    {
+        StageStats stageStats;
+        DistributedFragmentStats distributedFragmentStats = null;
+        if (stageInfo.isPresent()) {
+            stageStats = stageInfo.get().getStageStats();
+
+            double avgPositionsPerTask = stageInfo.get().getTasks().stream().mapToLong(task -> task.getStats().getProcessedInputPositions()).average().orElse(Double.NaN);
+            double squaredDifferences = stageInfo.get().getTasks().stream().mapToDouble(task -> Math.pow(task.getStats().getProcessedInputPositions() - avgPositionsPerTask, 2)).sum();
+            double sdAmongTasks = Math.sqrt(squaredDifferences / stageInfo.get().getTasks().size());
+
+            distributedFragmentStats = new DistributedFragmentStats(
+                    stageStats.getTotalCpuTime().convertToMostSuccinctTimeUnit(),
+                    stageStats.getTotalScheduledTime().convertToMostSuccinctTimeUnit(),
+                    formatPositions(stageStats.getProcessedInputPositions()),
```

Comment:
The DistributedFragmentRepresentation should contain unformatted values. That's up to the renderer to do when going from DistributedPlanRepresention -> textual form